File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>mp = new HashMap<>();
        int n = nums.length,num;
        for(int i=0;i<n;i++){
            num = nums[i];
            if(mp.containsKey(target-num))return new int[] {mp.get(target-num),i};
            mp.put(num,i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = -1, r = nums.size();
        while(l+1<r){
            int m = l + (r-l)/2;
            if(nums[m]<=target){
                l = m;
            }else{
                r = m;
            }
        }
        if(l>-1 and l<nums.size() and nums[l]==target)return l;
        return -1;
    }
};
